pandas/util/testing.py

```diff
@@ -1736,7 +1736,9 @@ def makeUnicodeIndex(k=10, name=None):
 def makeCategoricalIndex(k=10, n=3, name=None, **kwargs):
     """ make a length k index or n categories """
     x = rands_array(nchars=4, size=n)
-    return CategoricalIndex(np.random.choice(x, k), name=name, **kwargs)
+    return CategoricalIndex(
```

Comment:
Why this change?

Reply:
Primarily because this method is being used to generate some a `CategoricalIndex` in a parameterised fixture, which is then used as an input to `reindex`. If using `np.random.choice` this test will usually get a duplicate index and hence fail, while `np.arange` won't cause this issue because the test is asking for a short-enough index.

Note that most of the other `Index`-generating methods in this module (in particular `makeIntIndex`, `makeUIntIndex`) avoid creating duplicate indexes. Even `makeStringIndex`, `makeUnicodeIndex`, `makeFloatIndex` avoid creating duplicate indexes with high probability and `makeBoolIndex` seems to be trying to do the same thing if you ask for a short enough index.